### PR TITLE
Upgrade neo4j module to use features from v0.29.1 of testcontainers-go

### PR DESCRIPTION
**Pull request by danielorbach**
## What does this PR do?

This PR updates the implementation of neo4j enterprise to use the newly added `testcontainers.WithEnv` instead of directly accessing the environment of `req`, which is erroneous.

Also, this PR fixes a minor typo in the docs.

## Why is it important?

The `Env` map of `req` may be nil. Also, dog fooding our own API is a good practice.

## How to test this PR

I've tested that this PR does not break `neo4j` tests. I trust the CI other than that.

### CODE REVIEW

**Comment by danielorbach** [comment]
@mdelapenya appreciate your review 🙏🏿

File: docs/features/common_functional_options.md [hunk 1]
```diff
@@ -22,7 +22,7 @@ Using the `WithImageSubstitutors` options, you could define your own substitutio
 If you need to either pass additional environment variables to a container or override them, you can use `testcontainers.WithEnv` for example:
 
 ```golang
-postgres, err = postgresModule.RunContainer(ctx, testcontainers.WithEnv(map[string]string{"POSTGRES_INITDB_ARGS", "--no-sync"}))
+postgres, err = postgresModule.RunContainer(ctx, testcontainers.WithEnv(map[string]string{"POSTGRES_INITDB_ARGS": "--no-sync"}))
```
Comment (mdelapenya) [inline docs/features/common_functional_options.md:25]: Good catch!

**Review by mdelapenya — APPROVED** [review]
LGTM, thanks for cleaning up the code with the most recent features 👏👏👏

**Comment by mdelapenya** [comment]
The compose error comes from https://github.com/actions/runner-images/issues/9557

I'll work on updating the code tomorrow, then I'll merge this one.

Thanks!

**Pull request merged**